src/components/home/search/search.tsx

```diff
@@ -25,24 +21,22 @@ import { useStyles } from './styles';
 const text = appText.homePage.search;
 
 export function Search(): React.ReactNode {
+  const [searchField, setSearchField] = useState('');
   const navigate = useNavigate();
-  const { control, register } = useForm<ISearchFormData>();
   const { classes } = useStyles();
 
-  const textfieldEndButton = (
-    <Button
-      className={classes.textfieldEndButton}
-      colorVariant={ColorVariant.Light}
-      startIcon={<SearchIcon />}
-    >
-      <Typography variant="caption">{text.generate}</Typography>
-    </Button>
-  );
+  const filteredTemplates = TEMPLATES.filter((template) => {
+    return template.label.toLowerCase().includes(searchField.toLowerCase());
+  });
 
   function handleOpenTemplates(): void {
     navigate('/homepage/templates');
   }
 
+  function handleChange(e: { target: { value: SetStateAction<string> } }) : void {
```

Comment:
You can simply change to
function handleChange({ target: { value } }: React.ChangeEvent<HTMLInputElement>): void {
  setSearchField(value);
};